### ISSUE TITLE
Generating tqdm progress bar for non-interactive environments

### ISSUE THREAD

**Issue by lewismc** [zeehio/parmap#30]
This issue pertains to the limitation of being able to generate tqdm progress bars for parmap execution within interactive environments only.

The ability to generate tqdm progress bars in non-interactive environments e.g. docker containers, etc. would be extremely useful. If using parmap in a web application context, it would be really useful if progress could be written to log file(s).

In the above use case, [tqdm-loggable](https://pypi.org/project/tqdm-loggable/) could be leveraged. 

I'm interested to see what kind of response this issue gets before I attempt to create a PR for the feature.

Thanks for any feedback.

**Comment by Jaybuuum** [comment]
hello lewismc. You may be busy, but,
you must check My comments on this issue

- fswatch service cannot interpret more than one artifact at a time
- Implement DELETE's for UPDATE's to metadata only ingestion

Someone's stealing your GitHub and scamming it's. Several victims have already occurred.
You may suffer damage soon. It's a very serious situation.

**Comment by jay-buuum** [comment]
![Uploading IMG_3049.jpeg…]()

**Comment by jay-buuum** [comment]
![IMG_3050](https://github.com/zeehio/parmap/assets/138194055/15a1d5c2-53dc-4727-8fbc-0fc3ad52c233)
![IMG_3048](https://github.com/zeehio/parmap/assets/138194055/22ef0448-b2f6-47f4-b0c5-d1314531d666)
![IMG_3051](https://github.com/zeehio/parmap/assets/138194055/8d494a3b-b6f3-409c-8806-4df645ed8900)
![IMG_3052](https://github.com/zeehio/parmap/assets/138194055/225aa9b8-53b8-4f85-8288-b1f13cdbe52f)

**Comment by lewismc** [comment]
Comments above apart from my own are spam. The user has been reported. Please block the user from this project.

**Comment by jay-buuum** [comment]
No. I’m not strange people

**Comment by jay-buuum** [comment]
![IMG_3049](https://github.com/zeehio/parmap/assets/138194055/36cf110c-586a-4cfa-b9f2-6bdb1a32a5a6)
![IMG_3050](https://github.com/zeehio/parmap/assets/138194055/1dbd7554-94bb-45f5-bcd9-120232fa5457)
![IMG_3048](https://github.com/zeehio/parmap/assets/138194055/3e7cebc1-2557-4ce6-b52c-53dedc31e4bd)
![IMG_3051](https://github.com/zeehio/parmap/assets/138194055/2d57051c-ab57-4882-9d41-e23694d22a2b)
![IMG_3052](https://github.com/zeehio/parmap/assets/138194055/5302de1f-a044-4729-a9ae-b99fbeaac435)

**Comment by zeehio** [comment]
Reported and blocked. The noise is out. 
I hope those accounts stop harassing you soon.
We can now focus on the issue.

Thanks for all your comments. I understand what you would like to have.

From my side I would rather not depend on another package, even if optionally. However I believe I have a solution that may work for you:

The `pm_pbar` argument currently accepts two data types:
- `bool`: `True` or `False`, whether to show a progress bar or not
- `dict`: A dictionary with options to pass to `tqdm()`

My proposal would be to extend this argument accepting a `callable`:

- `callable`: A `tqdm`-compatible progress bar.

If you agree, I will implement it later this week.

**Comment by lewismc** [comment]
@zeehio your proposal sounds good. Thanks for offering to accommodate. Greatly appreciated. 👍

**Comment by zeehio** [comment]
Done and released in parmap 1.7.0, already in pypi and in conda-forge.

You can do this now:

```python
import parmap
from tqdm_loggable.auto import tqdm
parmap.map(print, range(10), pm_pbar = tqdm)
```

Or this if you want even more customizations:

```python
from functools import partial
import parmap
from tqdm_loggable.auto import tqdm
parmap.map(print, range(10), pm_pbar = partial(tqdm, desc = "example"))
```

Thanks for your suggestion!

**Comment by lewismc** [comment]
Absolutely fantastic @zeehio thank you for accommodating this request. Greatly appereciated.
